odoo/models.py

```diff
@@ -5379,22 +5380,32 @@ def convert_pgerror_unique(model, fields, info, e):
     if not ufields or model._table != table:
         return {'message': tools.ustr(e)}
 
+    del cr  # otherwise _() would get the closed cursor from above
```

Comment:
Alternatively: use something other than `cr` for the query above, or set `cr = model.cr`?